controllers/etcdcluster_controller_test.go

```diff
@@ -41,7 +41,7 @@ type StaticResponseMembersAPI struct {
 }
 
 func (s *StaticResponseMembersAPI) List(ctx context.Context) ([]etcdclient.Member, error) {
-	return s.parent.Members, nil
+	return s.parent.Members[:], nil
```

Comment:
Yeah, I didn't bother, because I was concentrating on fixing the race conditions, but I've now added a deep copy function to avoid any accidental mutation of the Member list.